### DISCUSSION TITLE
LLP/TCP Listener no longer receiving messages after upgrade.

### DISCUSSION THREAD

**Discussion by mr-shorten** [nextgenhealthcare/connect#4841] [Q&A] [answered]
Hello everyone,

Recently upgraded our Mirth Server from version 2.2.1 to 3.12 on Windows OS.

On Version 2.2.1 we had a LLP Listener configured as a server that was receiving out HL7 Messages. After upgrade, it combined into TCP Listener and we are not getting these messages anymore. i am not seeing any errors in the server or channel logs as to why this might be happening. I am a little clueless on were to start troubleshooting as i can still see these messages coming into the server via wireshark. 
 
Attached is both the 2,2,1 Channel and the Automatically Update channel for 3.12.
[2.2.1_CHANNEL.txt](https://github.com/nextgenhealthcare/connect/files/7508755/2.2.1_CHANNEL.txt)
[3.12_CHANNEL.txt](https://github.com/nextgenhealthcare/connect/files/7508756/3.12_CHANNEL.txt)
0

**Comment by pacmano1** [comment, marked as answer]
You probably need to look at the old listener and check port, framing and so on.

**Reply by mr-shorten** [reply]
Thanks for the quick response, everything seems to match best i can tell. The only difference that i think it might be, is in old mirth we had Record Separator Char and End Of Segment Char Configured as 0X0D. In New mirth we only see option to configure **Start of Message** and **End of Message** Characters could that maybe be the issue?

**Reply by pacmano1** [reply]
That would be the typical suspect.

**Reply by tonygermano** [reply]
I've never used mirth 2.x, but I think they are OK.
- "message start" is `0x0B` in both files.
- I don't know that "segment end" really means anything. I think that's referring to the HL7 segment delimiter.
- in the 2.x file you have "message end" as `0x1C` and "record separator" as `0x0D`, while in the 3.x file "message end" has them combined as `0x1C0D`.

I'm pretty sure those are both standard MLLP.

Do you see responses in Wireshark? What do you see on the dashboard? Does it show the channel connection connected or idle? What about the connection log at the bottom?

I wouldn't expect that you would see messages coming in wireshark unless a connection was established, but then I would either expect mirth to create a message or log an error that it couldn't.

**Reply by mr-shorten** [reply]
Apologies, i may be mistaken about the wireshark. The channel basically sits in Idle Status with no errors that i can see in the connection logs. I've tried various combinations of using the 0x1C both separated and combined with 0D but i honestly can't tell were the hold up is as everything was coming in fine in 2.x

**Reply by mr-shorten** [reply]
So after restarting several times i tried start from scratch, i imported the same channels from 2.x into a new 3.x instance and the channel started receiving. I am now getting this error however when trying to export from a channel reader/writer. A quick google search points to [this](https://forums.mirthproject.io/forum/mirth-connect/support/14857-replacement-for-messageobject-in-v3-3) page, but the solutions there did not seem to resolve. Apologies, i am new to mirth and have not been formally trained in this. 

`[2021-11-10 03:57:57,509]  ERROR  (com.mirth.connect.connectors.js.JavaScriptDispatcher:274): Error evaluating JavaScript Writer (JavaScript Writer "ImportPatient REST Service HTTPS" on channel 1b03c374-c7f8-4f81-8011-d8299ed89944).
com.mirth.connect.server.MirthJavascriptTransformerException: CHANNEL:
	TEST_PATIENT_IMPORT_OUTCONNECTOR:
	ImportPatient REST Service HTTPSSCRIPT SOURCE:
	JavaScript WriterSOURCE CODE:
	96:     }97: }98: function doScript() {99: var address = 'https://' + $('VALID_HOSTNAME_PROD') + '/Services/HL7/Patient.svc/ImportPatient'; 100: 101: var data = '<ImportPatient xmlns="http://afhcan.org/HL7/Patient/v1">'102:                 + '<xmlMessage><![CDATA[' + messageObject.getRawData() + ']]></xmlMessage>'103:                 + '<messageId>' + messageObject.getId() + '</messageId>'104:                 + '<messageSource>' + $('TEMPLATE_NAME') + '</messageSource>'105: + '</ImportPatient>';LINE NUMBER:
	101DETAILS:
	ReferenceError: "messageObject" is not defined.
	at a05920e0-56af-443b-a23b-bdb7dd4fdb55:101 (doScript)
	at a05920e0-56af-443b-a23b-bdb7dd4fdb55:154
	at com.mirth.connect.connectors.js.JavaScriptDispatcher$JavaScriptDispatcherTask.doCall(JavaScriptDispatcher.java:265)
	at com.mirth.connect.connectors.js.JavaScriptDispatcher$JavaScriptDispatcherTask.doCall(JavaScriptDispatcher.java:190)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:113)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)`

**Reply by pacmano1** [reply]
Post the actual channel export maybe?   From your release to the current release there are a slew of things to adjust.   The upgrade guides covered many of them.  Unfortunately, those guides seem to be hard to find.

Your release was 10 years ago.  It is  assuming a lot to think the upgrade would be seamless without your evaluating a lot of code in your channels for re-work.

**Reply by tonygermano** [reply]
`messageObject` I think was a 2.x thing that doesn't exist anymore. In a javascript writer, you would want to use `connectorMessage.getMessageId()` instead of `messageObject.getId()` to get the message id. In 3.x, message ids are long integers, only unique in the context of the channel in which they were created. I'm pretty sure in 2.x they were UUID values. So, you might want to consider if that will affect anything downstream, too.

Like @pacmano1 said, I would not expect a seamless upgrade with that big of a version jump.
